# KK1 notebook

## Imports och data loading

After downloading the data from seaborn, saving it locally to as csv file and reading from the local file.

In [38]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

penguins = sns.load_dataset("penguins")
penguins.to_csv("data/raw/penguins_raw.csv", index=False)

df = pd.read_csv("data/raw/penguins_raw.csv")

## Exploretory data analysis (EDA)

In [45]:
df.head()

,species,island,bill_length_mm,bill_depth_mm,flipper_length_mm,body_mass_g,sex
0,Adelie,Torgersen,39.1,18.7,181.0,3750.0,Male
1,Adelie,Torgersen,39.5,17.4,186.0,3800.0,Female
2,Adelie,Torgersen,40.3,18.0,195.0,3250.0,Female
3,Adelie,Torgersen,NaN,NaN,NaN,NaN,NaN
4,Adelie,Torgersen,36.7,19.3,193.0,3450.0,Female


## Dataset inspection

### Amount of rows and columns

In [44]:
df.shape


(344, 7)

### Datatypes 

In [43]:
df.info()


<class 'pandas.DataFrame'>
RangeIndex: 344 entries, 0 to 343
Data columns (total 7 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   species            344 non-null    str    
 1   island             344 non-null    str    
 2   bill_length_mm     342 non-null    float64
 3   bill_depth_mm      342 non-null    float64
 4   flipper_length_mm  342 non-null    float64
 5   body_mass_g        342 non-null    float64
 6   sex                333 non-null    str    
dtypes: float64(4), str(3)
memory usage: 18.9 KB


### Quick summry of the values in the data set

In [42]:
df.describe()


,bill_length_mm,bill_depth_mm,flipper_length_mm,body_mass_g
count,342.000000,342.000000,342.000000,342.000000
mean,43.921930,17.151170,200.915205,4201.754386
std,5.459584,1.974793,14.061714,801.954536
min,32.100000,13.100000,172.000000,2700.000000
25%,39.225000,15.600000,190.000000,3550.000000
50%,44.450000,17.300000,197.000000,4050.000000
75%,48.500000,18.700000,213.000000,4750.000000
max,59.600000,21.500000,231.000000,6300.000000


### Missing values 

In [41]:
df.isna().sum()

species               0
island                0
bill_length_mm        2
bill_depth_mm         2
flipper_length_mm     2
body_mass_g           2
sex                  11
dtype: int64

## Data cleaning

 Dropping all the nan values because even after dropping it has still a good chuck left to experement with. i went from 344 to 333 rows.  
 the missing values in dataset is probably because of incomple messurments during data collection. 


In [48]:
df_clean = df.dropna()

### Amount of duplicate data in the dataset

In [47]:
df_clean.duplicated().sum()

np.int64(0)

### Checking the transition of raw and clean data

In [52]:
print("raw data: ", df.shape)
print("clean data: ", df_clean.shape)

raw data:  (344, 7)
clean data:  (333, 7)
